.github/workflows/ci.yml

```diff
@@ -7,6 +7,22 @@ on:
     branches: [ main ]
 
 jobs:
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+
+      - name: Install required packages
+        run: pip install ruff
```

Comment:
Should this be in poetry's dev dependencies, and get run on commit? Maybe we can add a pre-commit hook if one doesn't exist already.